api/v1alpha4/types.go

```diff
@@ -322,6 +322,22 @@ type AzureSharedGalleryImage struct {
 	// time even if a new version becomes available.
 	// +kubebuilder:validation:MinLength=1
 	Version string `json:"version"`
+	// Publisher is the name of the organization that created the image
+	// +optional
+	Publisher *string `json:"publisher,omitempty"`
```

Comment:
I understand what you meant by needing those fields now... For Marketplace image we could reuse the same values for the image but here they are distinct values for the image that was used as source. In that case, it makes more sense to get rid of thirdPartyImage and make it more explicit that tthe Publisher/Offfer/SKU are for generating the plan and should be the Publisher/Offer/SKU of the image that was used as base to create the SIG image.

Reply:
You mean explicit in the comments or where?
Getting rid of the flag.

Reply:
I meant in the comments, this looks good.

Can you also add a note to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/book/src/topics/custom-images.md#using-shared-image-gallery-recommended ?

Reply:
Done